Platform/RaspberryPi/Drivers/Genet/Genet.c

```diff
@@ -0,0 +1,132 @@
+/** @file
+
+  Copyright (c) 2019, Jeremy Linton All rights reserved.<BR>
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  This driver acts like a stub to set the BCM 
+  Genet MAC address, until the actual network driver
+  is in place. This should allow us to retrieve the
+  mac address directly from the hardware in supported
+  OS's rather than passing it via DSDT (which isn't 
+  ideal for a number of reasons, foremost the hardware
+  should be self describing).
+
+**/
+
+#include <Library/ArmLib.h>
+#include <Library/DebugLib.h>
+#include <Library/DevicePathLib.h>
+#include <Library/IoLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+
+#include <PiDxe.h>
+#include <Protocol/RpiFirmware.h>
+
+#define GENET_BASE                 0xfd580000  // len = 0x10000
```

Comment:
I think we'll want a PCD here to be in line with what we're doing elsewhere.

And we probably also want to move all the defines into a separate header, especially if we're going to create a full blown Genet driver.

As a matter of fact, considering that Genet is not exclusive to Pi platforms (there already appear to exist 5 revisions of Genet used in various hardware implementations that have nothing to do with RPi), I'd push to move this driver to `Silicon/Broadcom/`  instead of having it in `Platforms/RaspberryPi/`, similar to what we have for RNG (though I don't think we want to attach this one to `Bmc2711` or `Bcm283x` so we'll probably need to create a new `Silicon/Broadcom/Drivers/Net/BcmGenetDxe/`, [like what Marvell has](https://github.com/tianocore/edk2-platforms/tree/master/Silicon/Marvell/Drivers/Net)). 

At any rate, I more or less expect any EDK2 reviewer/maintainer with knowledge that Genet is not a Pi specific thing to ask us to move our driver into Silicon during the review process, especially if we hint that we're going to build on this moving forward.

Unless there are objections then, I'll be working on creating a new pull request that does just that (and of course I'll be applying Ard's remarks as I'm doing so).

Reply:
Funny, I have the driver I was working in Silicon, but this seemed like a temp solution.. 

Sure, I'm good with whatever, I can apply Ard's suggestions, if your short on time.

Reply:
I've already applied them in what I'm working on. Just need to move files around now. I'm hoping to have something in a few hours, but that's dependent on nothing else pre-empting this.